generator/mwm_diff/diff.cpp

```diff
@@ -0,0 +1,56 @@
+#include "generator/mwm_diff/diff.hpp"
+
+#include "coding/file_reader.hpp"
+#include "coding/file_writer.hpp"
+#include "coding/read_write_utils.hpp"
+#include "coding/reader.hpp"
+
+#include "base/logging.hpp"
+
+using namespace std;
+
+namespace
+{
+// We could use my::CopyFileX but it fails if the file at |srcFile| is empty.
+bool CopyFile(string const & srcFile, string const & dstFile)
+{
+  size_t constexpr kBufferSize = 1024 * 1024;
```

Comment:
This buffer size is too large, usual value is about 4-8-16 kbytes. Consider to use st_blksize from stat structure, or at least hardcode the constant to 8 kbytes.